### ISSUE TITLE
Text for CS1673 needs updating for local methods

### ISSUE THREAD

**Issue by jskeet** [dotnet/roslyn#19986]
**Version Used**:  2.2.0.61624

**Steps to Reproduce**:

Code:

```csharp
using System;

public struct Foo
{
    public string name;
    
    public void PrintName()
    {
        PrintNameImpl();
        void PrintNameImpl()
        {
            Console.WriteLine(name);
        }
    }
}
```

**Expected Behavior**:

Initially, I expected this to compile. I'm okay with it not compiling, but the error message should be accurate, explaining that there's a restriction around local methods.

**Actual Behavior**:

Error message, emphasis mine:

> Test.cs(12,31): error CS1673: **Anonymous methods, lambda expressions, and query expressions** inside structs cannot access instance members of 'this'. Consider copying 'this' to a local variable outside the anonymous method, lambda expression or query expression and using the local instead.

There's no anonymous method, lambda expression or query expression here. Unless there's some compatibility issue with updating the error message, it should just be updated to include local methods.

**Comment by jcouv** [comment]
FYI @agocke for triage.

**Comment by kevinsun-dev** [comment]
Fixed in #44835 .
